packages/tests/tests/peer_exchange.node.spec.ts

```diff
@@ -137,4 +139,68 @@ describe("Peer Exchange", () => {
       expect(waku.libp2p.peerStore.has(await nwaku2.getPeerId())).to.be.true;
     });
   });
+
+  describe("compliance test", async function () {
+    this.timeout(25_000);
+
+    let waku: LightNode;
+    let nwaku1: Nwaku;
+    let nwaku2: Nwaku;
+
+    tests({
+      async setup() {
+        nwaku1 = new Nwaku("1");
+        nwaku2 = new Nwaku("2");
+        await nwaku1.start({
+          discv5Discovery: true,
+          peerExchange: true,
+        });
+
+        const enr = (await nwaku1.info()).enrUri;
+
+        await nwaku2.start({
+          discv5Discovery: true,
+          peerExchange: true,
+          discv5BootstrapNode: enr,
+        });
+
+        waku = await createLightNode();
+
+        await waku.start();
+        const nwaku2Ma = await nwaku2.getMultiaddrWithId();
+
+        setTimeout(async () => {
+          await waku.libp2p.dialProtocol(nwaku2Ma, PeerExchangeCodec);
+          await new Promise<void>((resolve) => {
+            waku.libp2p.peerStore.addEventListener(
+              "change:protocols",
+              (evt) => {
+                if (evt.detail.protocols.includes(PeerExchangeCodec)) {
+                  resolve();
+                }
+              }
+            );
+          });
+        }, 1000);
+
+        // the ts-ignores are added ref: https://github.com/libp2p/js-libp2p-interfaces/issues/338#issuecomment-1431643645
+        const components = {
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          connectionManager: waku.libp2p.connectionManager,
+          peerStore: waku.libp2p.peerStore,
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
```

Comment:
Can you solve it by force type casting?

Reply:
Yes, but readability can be improved by making a tmp variable 
`const libp2p = waku.libp2p as Libp2pComponents;`

Reply:
Great! It can be even improved further 
```ts
const components = {
  connectionManager,
  registrar,
  peerStore,
};

This is just a nit.